common/src/chain/config/builder.rs

```diff
@@ -107,14 +102,40 @@ impl ChainType {
                                 PerThousand::new(1).expect("must be valid"),
                                 PoSConsensusVersion::V1,
                             ),
-                        }),
+                        },
                     ),
                 ];
                 NetUpgrades::initialize(upgrades).expect("net upgrades")
             }
             ChainType::Signet => NetUpgrades::unit_tests(),
         }
     }
+
+    fn default_chainstate_upgrades(&self) -> NetUpgrades<ChainstateUpgrade> {
+        match self {
+            ChainType::Mainnet | ChainType::Regtest | ChainType::Signet => {
+                let upgrades = vec![(
+                    BlockHeight::new(0),
+                    ChainstateUpgrade::new(TokenIssuanceVersion::V1),
+                )];
+                NetUpgrades::initialize(upgrades).expect("net upgrades")
+            }
+            ChainType::Testnet => {
```

Comment:
I think there's a misunderstanding on how this should work. We shouldn't have multiple upgrade vectors. We should have one upgrade vector with different things happening. Something like:

At height 0: Genesis
At height 1: PoS
At height 10000: TokenV1 + PoSv2

And so on.

Reply:
I think that's more complicated to work with. It would look like:
```
struct Upgrade {
    consensus_upgrade: Option<ConsensusUpgrade>,
    chainstate_upgrade: Option<ChainstateUpgrade>,
}
```

where one or both values must always be set. And after that we will have to decompose this `Upgrage` struct to get particular values every time we want to check something. I don't see the point.

Reply:
We don't need to have separate structs with different types of upgrades. We need one struct that's monotonic with block height. There's only one set of upgrades that happen in order that affect all the code in all nets. The only difference is the heights.

Reply:
Maybe I didn't describe my idea clearly. This `struct Upgrade` (the name can isn't final) is the one struct that goes to `NetUpgrade<Upgrade>` collection. In this way it's both monotonic with height and separates chainstate upgrades from consensus.
I first implemented it but didn't like the way it complicated code so I thought it's better to make separate netupgarde collections. Mainly because it seems cleaner to me in both managing the code and understanding the upgrades.

Reply:
As was discussed the whole NetUpgrade approach has to be reconsidered. I'll leave the code as it is in the PR but will rework it as a separate task right after this one.